src/messageschannel.cpp

```diff
@@ -119,12 +123,19 @@ void MatrixMessagesChannel::onPendingEventChanged(int pendingEventIndex)
     header[QStringLiteral("message-sender-id")] = QDBusVariant(m_targetId);
     header[QStringLiteral("message-type")]      = QDBusVariant(Tp::ChannelTextMessageTypeDeliveryReport);
     header[QStringLiteral("delivery-status")]   = QDBusVariant(tpDeliveryStatus);
-    header[QStringLiteral("delivery-token")]    = QDBusVariant(pendingEvent.event()->transactionId());
+    header[QStringLiteral("delivery-token")]    = QDBusVariant(pendingEvent.event()->id());
     partList << header;
 
     addReceivedMessage(partList);
 }
 
+void MatrixMessagesChannel::sendTypingNotification(bool isTyping)
```

Comment:
I would suggest to wire Tp::ChannelChatState down to this interface so we'll have less to refactor once Matrix will gain something more than trivial `bool` state support.
Name `sendChatStateNotification()` would fit the new signature. :)